### PR TITLE
SAK-33574 : added cache to ExternalHierarchyLogicImpl::getNodeById

### PR DESCRIPTION
**Pull request by frasese**

### CODE REVIEW

**Comment by mpellicer** [comment]
@frasese 

I think we should add the option to define the cache TTL.

**Comment by frasese** [comment]
We have noticed the compilation errors. They are involved with tests. We are going to solve that.

File: sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/ExternalHierarchyLogicImpl.java [hunk 1]
```diff
@@ -342,6 +358,14 @@ public EvalHierarchyNode getRootLevelNode() {
      * @see org.sakaiproject.evaluation.providers.EvalHierarchyProvider#getNodeById(java.lang.String)
      */
     public EvalHierarchyNode getNodeById(String nodeId) {
+        if(cache.containsKey(nodeId)){
```
Comment (ottenhoff) [inline sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/ExternalHierarchyLogicImpl.java]: I believe the preferred method is to do one cache.get(nodeId) and then check to see if that result is null. I believe containsKey should be avoided.

File: sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/ExternalHierarchyLogicImpl.java [hunk 2]
```diff
@@ -94,6 +96,18 @@ public void setAuthzGroupService( AuthzGroupService authzGroupService ) {
...
+    private MemoryService memoryService;
+    public void setMemoryService(MemoryService memoryService){
+    	this.memoryService = memoryService;
+    }
+    
+    private Cache cache;
+    public void setCache(Cache cache) {
+		this.cache = cache;
+	}
+    private final String CACHE_NAME = "org.sakaiproject.evaluation.cache";
```
Comment (ottenhoff) [inline sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/ExternalHierarchyLogicImpl.java]: Can you name the cache org.sakaiproject.hierarchy.nodes instead?

**Comment by frasese** [comment]
Done

**Pull request merged**